### PR TITLE
Update execute_model to support async scheduling in vllm

### PR DESCRIPTION
**Pull request by sixiang-google**
# Description

https://github.com/vllm-project/vllm/pull/26866 yields None in some of the future model output. Change the model output handling to accommodate this. 

b/458937786

Also: bug fix in inconsistent type in/out kv cache transfer

# Tests

pytest -v tests/core/test_core_tpu.py

# Checklist

Before submitting this PR, please make sure:
- I have performed a self-review of my code.
- I have necessary comments in my code, particularly in hard-to-understand areas.
- I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

File: tpu_inference/runner/kv_cache_manager.py [hunk 1]
```diff
@@ -389,8 +389,8 @@ def get_sharding(x):
 
             sharding_spec_pytree = jax.tree.map(get_sharding, kv_cache_slices)
             transferred_kv_cache = experimental_reshard.reshard(
-                tuple(kv_cache_slices),
-                tuple(sharding_spec_pytree),
+                kv_cache_slices,
```
Comment (QiliangCui) [inline tpu_inference/runner/kv_cache_manager.py:392]: Is this related to the async scheduler change?

**Review by QiliangCui — APPROVED** [review]

File: tpu_inference/runner/kv_cache_manager.py (hunk 1 above)
Comment (sixiang-google) [reply tpu_inference/runner/kv_cache_manager.py:392]: No. Added in description
